packages/tlsync/src/lib/TLServer.ts

```diff
@@ -78,14 +79,24 @@ export abstract class TLServer {
 				}
 			}
 
-			// If we still don't have a room, create a new one
+			// If we still don't have a room, throw an error.
 			if (roomState === undefined) {
-				roomOpenKind = 'new'
-
-				roomState = {
-					persistenceKey,
-					room: new TLSyncRoom(this.schema),
-				}
+				// This is how it bubbles down to the client:
+				// 1.) From here, it's caught in the catch block of handleConnection,
+				//   and we return `room_not_found` to TLDrawDurableObject.
+				// 2.) In TLDrawDurableObject, we return a 404 to the client, we accept and
+				//   and then immediately close the client. This lets us send a TLCloseEventCode.NOT_FOUND
+				//   closeCode down to the client.
+				// 3.) joinExistingRoom which handles the websocket upgrade is not affected.
+				//   Again, we accept the connection, it's just that we immediately close right after.
+				// 4.) In ClientWebSocketAdapter, ws.onclose is called, and that calls _handleDisconnect.
+				// 5.) _handleDisconnect sets the status to 'error' and calls the onStatusChange callback.
+				// 6.) On the dotcom app in useRemoteSyncClient, we have socket.onStatusChange callback
+				//   where we set TLIncompatibilityReason.RoomNotFound and close the client + socket.
+				// 7.) Finally on the dotcom app we use StoreErrorScreen to display an appropriate msg.
+				//
+				// Phew!
+				throw new Error('room_not_found')
```

Comment:
Would probably just return something here instead of throwing. It's expected that this will happen and not something out of the ordinary.

Reply:
Seems you already updated this, resolving 😄